src/assign.c

```diff
@@ -458,7 +458,7 @@ SEXP assign(SEXP dt, SEXP rows, SEXP cols, SEXP newcolnames, SEXP values)
     coln--;
     SEXP thisvalue = RHS_list_of_columns ? VECTOR_ELT(values, i) : values;
     vlen = length(thisvalue);
-    if (isNull(thisvalue) && !isNull(rows)) error(_("When deleting columns, i should not be provided"));  // #1082, #3089
+    if (isNull(thisvalue) && !isNull(rows)) error(_("Direct assignment of NULL when `i` is provided is not supported. If you intend to assign NULL to a column, use `NULL` without `i`. For list columns, if you intend to assign NULL to a specific item, use `list(list(NULL))` in your query."));  // #1082, #3089
```

Comment:
please remove backticks in string literals.
only use backticks in markdown on github issues/prs (not in code in git repo)

Reply:
done 👍

Reply:
can you please checkout if any more changes are required for this PR.

Reply:
Hi Nitish, sorry for the delay. From my perspective, we are basically in code freeze until 1.16.0 is completed+released. Appreciate your patience for the next week or two, thanks!

Reply:
sure, thanks for info